cmd/provider.go

```diff
@@ -23,56 +19,46 @@ var providerCmd = Cmd{
 func runProvider(args ...string) {
 	fs := flag.NewFlagSet("provider", flag.ExitOnError)
 	addrFlag := fs.String("addr", "", "address to listen at (host:port)")
+	localFlag := fs.String("local", "", "address to listen at for local provider api (host:port)")
 	fs.Parse(args)
 
 	homedir, err := findHomeDir()
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	p, err := provider.LoadFromDisk(path.Join(homedir, "provider"))
+	logfile, err := os.OpenFile(path.Join(homedir, "provider", "provider.log"), os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("Cannot create log file: %s", err)
 	}
+	defer logfile.Close()
+	logger := log.New(logfile, "", log.LstdFlags)
 
-	addr := *addrFlag
-	if len(addr) == 0 {
-		addr = p.Config.ApiAddr
+	p, err := provider.InitProvider(homedir)
+	if err != nil {
+		log.Fatalf("Failed to init provider: %s", err)
 	}
 
-	// Register with metadata service.
-	if !p.Config.IsRegistered {
-		pubKeyBytes, err := ioutil.ReadFile(p.Config.PublicKeyFile)
-		if err != nil {
-			log.Fatal("Could not read public key file. Error: ", err)
-		}
-		info := core.ProviderInfo{
-			PublicKey:  string(pubKeyBytes),
-			Addr:       addr,
-			SpaceAvail: 1 << 32,
-		}
-		metaService := metaserver.NewClient(p.Config.MetaAddr, &http.Client{})
-		config, err := metaService.RegisterProvider(&info)
-		if err != nil {
-			log.Fatalf("Unable to register with metaservice. Error: %s", err)
-		}
-		p.Config.ProviderID = config.ID
-		p.Config.IsRegistered = true
-		err = util.SaveJson(path.Join(homedir, "provider", "config.json"), p.Config)
-		if err != nil {
-			log.Fatalf("Unable to update config after registering with metaserver. Error: %s", err)
-		}
+	addr := *addrFlag
+	localAddr := *localFlag
+	// only run local facing server (statistics for ui) when localAddr is set
```

Comment:
I might update this to always run the local server. We should probably add a configuration field for this address and set a default.

Reply:
A static always run default was my first thought. I mostly just added this because tests with multiple providers would all try to bind the same port for local server. 

Also it seemed mostly relevant for the UI aspect of provider and we could easily just use a default for our kickoff from the GUI.

I am open to the config option too I'm just unsure if it would be worth adding to the UI.